pkg/network/protocols/postgres/protocol.go

```diff
@@ -0,0 +1,185 @@
+}
+
+// ConfigureOptions add the necessary options for the postgres monitoring to work, to be used by the manager.
+func (p *protocol) ConfigureOptions(mgr *manager.Manager, opts *manager.Options) {
+	opts.MapSpecEditors[inFlightMap] = manager.MapSpecEditor{
+		MaxEntries: p.cfg.MaxUSMConcurrentRequests,
+		EditorFlag: manager.EditMaxEntries,
+	}
+	utils.EnableOption(opts, "postgres_monitoring_enabled")
+	// Configure event stream
+	events.Configure(p.cfg, eventStream, mgr, opts)
+}
+
+// PreStart runs setup required before starting the protocol.
+func (p *protocol) PreStart(mgr *manager.Manager) (err error) {
+	p.eventsConsumer, err = events.NewConsumer(
+		eventStream,
+		mgr,
+		p.processPostgres,
+	)
+	if err != nil {
+		return
+	}
+
+	p.eventsConsumer.Start()
+
+	return
+}
+
+// PostStart starts the map cleaner.
+func (p *protocol) PostStart(mgr *manager.Manager) error {
+	// Setup map cleaner after manager start.
+	p.setupMapCleaner(mgr)
+	return nil
+}
+
+// Stop stops all resources associated with the protocol.
+func (p *protocol) Stop(*manager.Manager) {
+	// mapCleaner handles nil pointer receivers
+	p.mapCleaner.Stop()
+
+	if p.eventsConsumer != nil {
+		p.eventsConsumer.Stop()
+	}
+}
+
+// DumpMaps dumps map contents for debugging.
+func (p *protocol) DumpMaps(w io.Writer, mapName string, currentMap *ebpf.Map) {
+	if mapName == inFlightMap { // maps/postgres_in_flight (BPF_MAP_TYPE_HASH), key ConnTuple, value EbpfTx
+		var key netebpf.ConnTuple
+		var value EbpfTx
+		protocols.WriteMapDumpHeader(w, currentMap, mapName, key, value)
+		iter := currentMap.Iterate()
+		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
+			spew.Fdump(w, key, value)
+		}
+	}
+}
+
+// GetStats returns a map of Postgres stats.
+func (p *protocol) GetStats() *protocols.ProtocolStats {
+	p.eventsConsumer.Sync()
+
+	return &protocols.ProtocolStats{
+		Type:  protocols.Postgres,
+		Stats: p.statskeeper.GetAndResetAllStats(),
+	}
+}
+
+// IsBuildModeSupported returns always true, as postgres module is supported by all modes.
+func (*protocol) IsBuildModeSupported(buildmode.Type) bool {
+	return true
+}
+
+func (p *protocol) processPostgres(events []EbpfEvent) {
+	for i := range events {
+		tx := &events[i]
+		p.statskeeper.Process(&EventWrapper{EbpfEvent: tx})
+	}
+}
+
+func (p *protocol) setupMapCleaner(mgr *manager.Manager) {
+	postgresInflight, _, err := mgr.GetMap(inFlightMap)
+	if err != nil {
+		log.Errorf("error getting %s map: %s", inFlightMap, err)
+		return
+	}
+	mapCleaner, err := ddebpf.NewMapCleaner[netebpf.ConnTuple, EbpfTx](postgresInflight, 1024)
+	if err != nil {
+		log.Errorf("error creating map cleaner: %s", err)
+		return
+	}
+
+	// Clean up idle connections. We currently use the same TTL as HTTP, but we plan to rename this variable to be more generic.
+	ttl := p.cfg.HTTPIdleConnectionTTL.Nanoseconds()
+	mapCleaner.Clean(p.cfg.HTTPMapCleanerInterval, nil, nil, func(now int64, key netebpf.ConnTuple, val EbpfTx) bool {
```

Comment:
This is a bit out of scope, but I think we should change this variable name. We are using this interval time in different protocols, and it is always based on 'HTTP'. That seems odd to me. If you agree, I can take care of it

Reply:
true